api/interestr/api/tests.py

```diff
@@ -372,14 +409,88 @@ def test_create_template(self):
         self.client.force_authenticate(user=self.test_user)
         template_fields = {
             'name': 'Basic template',
-            'fields': ['short_text', 'long_text'],
+            'fields': [{
+                'type': 'text',
+                'legend': 'Title',
+                'inputs': [
+                    {
+                        'type': 'text'
+                    }
+                ]
+            }],
             'group': self.test_group.id
         }
         response = self.client.post('/api/v1/data_templates/',
                                     template_fields, format='json')
         self.assertEqual(response.status_code, 201,
                          responseError(response, 'Create Data Template'))
 
+    def test_create_wrong_format_template(self):
+        self.client.force_authenticate(user=self.test_user)
+        template_fields = {
+            'name': 'Wrong Basic Template',
+            'fields': [{
+                'wrong': 'wrong'
+            }],
+            'group': self.test_group.id
+        }
+
+        response = self.client.post('/api/v1/data_templates/',
+                                    template_fields, format='json')
+        self.assertEqual(response.status_code, 400,
+                         responseError(response, 'Create Wrong Format Data Template', False))
+
+    def test_create_wrong_type_template(self):
+        self.client.force_authenticate(user=self.test_user)
+        template_fields = {
+            'name': 'Basic template',
+            'fields': [{
+                'type': 'wrong',
+                'legend': 'Title',
+                'inputs': [
+                    {
+                        'type': 'text'
+                    }
+                ]
+            }],
+            'group': self.test_group.id
+        }
+        response = self.client.post('/api/v1/data_templates/',
+                                    template_fields, format='json')
+        self.assertEqual(response.status_code, 400,
+                         responseError(response, 'Create Wrong Type Data Template', False))
+
+    def test_create_multiple_field_same_legend_template(self):
+        self.client.force_authenticate(user=self.test_user)
+        template_fields = {
+            'name': 'Basic template',
+            'fields': [
+                {
+                    'type': 'text',
+                    'legend': 'SameLegend',
+                    'inputs': [
+                        {
+                            'type': 'text'
+                        }
+                    ]
+                },
+                {
+                    'type': 'text',
+                    'legend': 'SameLegend',
+                    'inputs': [
+                        {
+                            'type': 'text'
+                        }
+                    ]
+                }
```

Comment:
It could be better if the second field had the same legend (already ok), but with a completely different structure, i.e. different type and inputs. It is not clear whether this test fails for the fields being completely identical, or only because their legends are the same.

Reply:
The test is sufficient. If we have to test if our system allows multiple fields with completely identical fields that only have different legends, we should write a different test case for that.

Reply:
The title `test_create_multiple_field_same_legend_template` misleads the reviewer that this test is for testing specifically for the identical field legends. As it is, it could very well be testing (and failing) for having two fields with identical types.

Reply:
> It could very well be testing (and failing) for having two fields with identical types.

Why?

Reply:
Let me put it this way: How do we know that we get a status code of 400 because the field legends are the same, and not for another reason?

Reply:
We do not know. And probably will never if we do not add another case I mentioned. You cannot test everything at once.

Reply:
This test is too broad. It can be made more specific to honor its name, to a degree that I cannot anymore complain about.

I mean, either change the name of the test into `test_create_multiple_identical_field_template`, or one of the two fields' type and inputs into something else. The name of the test and the test itself are not in accordance as it is now.